README.md

```diff
@@ -1,2 +1,257 @@
+
+- It checks if the `structure` argument is a pointer to a struct. If not, it returns an error.
+
+- It identifies the primary key field in the provided struct using the "primarykey" tag.
+
+- It constructs an SQL query to retrieve the first record from the specified `tableName` based on the primary key.
+
+- The query results are scanned into the provided `structure`, which should be a pointer to a struct.
+
+### Example Usage:
+
+```go
+var result MyStruct
+err := FindFirst("your_table", &result)
+if err != nil {
+    // Handle the error
+}
+```
+
+## FindLast Function
+
+The `FindLast` function retrieves the last record from a specified database table based on the primary key field of a provided structure. This function is particularly useful when you want to retrieve the row with the largest primary key value from a table.
+
+### Parameters:
+
+- `tableName` (string): The name of the database table from which to retrieve the record.
+- `structure` (interface{}): A pointer to a struct (e.g., &MyStruct{}) representing the structure into which the retrieved data will be scanned. The primary key field of the struct is used to determine the record to retrieve.
+
+### Return Value:
+
+- `error`: An error is returned if any of the following conditions are met:
+  - The provided structure is not a pointer to a struct.
+  - The primary key field is not found in the struct or is missing the "primarykey" tag.
+  - There is an issue with executing the SQL query or scanning the result.
+
+### Functionality:
+
+- The `FindLast` function queries the specified database table to retrieve the last record based on the primary key field of the provided structure.
+
+- It checks if the `structure` argument is a pointer to a struct. If not, it returns an error.
+
+- It identifies the primary key field in the provided struct using the "primarykey" tag.
+
+- It constructs an SQL query to retrieve the last record from the specified `tableName` based on the primary key.
+
+- The query results are scanned into the provided `structure`, which should be a pointer to a struct.
+
+### Example Usage:
+
+```go
+var result MyStruct
+err := FindLast("your_table", &result)
+if err != nil {
+    // Handle the error
+}
+```
+## Find Function
+
+The `Find` function queries a database table based on the provided criteria and scans the result into a slice of structs.
+
+### Parameters:
+
+- `data` (`map[string]interface{}`): A map containing the query criteria.
+  - `tablename` (string, required): The name of the database table to query.
+  - `columnname` (string, required): The name of the column to filter on.
+  - `columnvalue` (interface{}, required): The value to filter the `columnname` by.
+  - `sortcolumn` (string, optional): The name of the column to sort the results by. Defaults to the primary key of the table defined in the structure.
+  - `sortvalue` (string, optional): The sorting order, which can be "ASC" (ascending) or "DESC" (descending). Defaults to "ASC".
```

Comment:
i have a some suggestion for this 
- can we need to make default order `DESC` because find last is provide last value of database table
- can we make 'columnname', columnvalue also optional (set * on column name, or primary key column name) when developer not set its value 


this same terminology can we apply on update and delete ?

Reply:
this find function will need to find in which column you are finding your query so its compulsory to provide the columnname and value
yes we can make the order desc its asc by default for getting result in default formate like sql